service/mantle/fedex/FedexServices.xml

```diff
@@ -0,0 +1,112 @@
+
+    <service verb="validate" noun="PostalAddress">
+            <implements service="mantle.shipment.CarrierServices.validate#ShippingPostalAddress"/>
+        <actions>
+            <!-- If a party/facility is expired(If replacesContactMechId is set) then address is not validated -->
+            <if condition="partyId"><then>
+                <entity-find entity-name="mantle.party.contact.PartyContactMechInfo" list="replacesPcmiList">
+                    <date-filter/><econdition field-name="partyId"/>
+                    <econdition field-name="replacesContactMechId" from="contactMechId"/>
+                    <select-field field-name="contactMechId"/><order-by field-name="contactMechId"/>
+                </entity-find>
+                <if condition="replacesPcmiList"><set field="contactMechId" from="replacesPcmiList[0].contactMechId"/>
+                    <return/></if>
+            </then><else-if condition="facilityId">
+                <entity-find entity-name="mantle.facility.FacilityContactMechInfo" list="replacesFcmiList">
+                    <date-filter/><econdition field-name="facilityId"/>
+                    <econdition field-name="replacesContactMechId" from="contactMechId"/>
+                    <select-field field-name="contactMechId"/><order-by field-name="contactMechId"/>
+                </entity-find>
+                <if condition="replacesFcmiList"><set field="contactMechId" from="replacesFcmiList[0].contactMechId"/>
+                    <return/></if>
+            </else-if></if>
+            <!-- Searching FedexApiToken -->
+            <entity-find-one entity-name="ShippingGatewayOption" value-field="apiTokenOpt">
+                    <field-map field-name="shippingGatewayConfigId" from="shippingGatewayConfigId"/>
+                    <field-map field-name="optionEnumId" value="FedexApiToken"/>
+            </entity-find-one>
+            <if condition="!apiTokenOpt.optionValue">
+                <return type="warning" message=" ${shippingGatewayConfigId} has no FedexApiToken, not validating address"/>
+            </if>
+            <!-- Converting Address Data to JSON Request Map -->
+            <service-call name="mantle.fedex.FedexServices.get#AddressMap" in-map="[contactMechId:contactMechId]"
+                out-map="addressMap"/>
+            <if condition="!addressMap"><return/></if>
+
+            <!-- Setting Connection With Fedex -->
+            <if condition="addressMap.isAddressComplete=='true'">
+            <service-call name="mantle.fedex.FedexServices.send#FedexRequest" out-map="response"
+                in-map="[shippingGatewayConfigId:shippingGatewayConfigId,method:'POST', path:'address/v1/addresses/resolve',
+                contentType:'application/json', requestMap:addressMap.addressMap, apiTokenValue:apiTokenOpt.optionValue]"/>
+            </if>
+
+            <!-- Verifying Correctness of Address -->
+            <set field="responseMap" from="response.responseMap"/>
```

Comment:
If addressMap.isAddressComplete is false, this is potential NPE as the control flow reaches this code. Please check for proper handling.